contracts/libraries/Math.sol

```diff
@@ -80,6 +97,288 @@ library Math {
+                    (numerator - 1701326769683909545783398326890970) /
+                    17013267696839095457833983268909703;
+            }
+
+            if (tick == tickDown) {
+                return int24(tick);
+            } else {
+                if (sqrtPriceX96 < getSqrtRatioAtTick(int24(tick))) {
+                    tick -= 1;
+                }
+                return int24(tick);
+            }
+        }
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                            LIQUIDITY AMOUNTS (STRIKE+WIDTH)
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice Calculates the amount of token0 received for a given liquidityChunk
+    /// @dev Had to use a less optimal calculation to match Uniswap's implementation
+    /// @param liquidityChunk variable that efficiently packs the liquidity, tickLower, and tickUpper.
+    /// @return amount0 The amount of token0
+    function getAmount0ForLiquidity(
+        uint256 liquidityChunk
+    ) internal pure returns (uint256 amount0) {
+        uint160 lowPriceX96 = getSqrtRatioAtTick(liquidityChunk.tickLower());
+        uint160 highPriceX96 = getSqrtRatioAtTick(liquidityChunk.tickUpper());
+
+        return
+            mulDiv(
+                uint256(liquidityChunk.liquidity()) << 96,
+                highPriceX96 - lowPriceX96,
+                highPriceX96
+            ) / lowPriceX96;
+
+        //return (uint256(liquidityChunk.liquidity()) << 96) / lowPriceX96 -  (uint256(liquidityChunk.liquidity()) << 96) / highPriceX96;
+    }
+
+    /// @notice Calculates the amount of token1 received for a given liquidityChunk
+    /// @param liquidityChunk variable that efficiently packs the liquidity, tickLower, and tickUpper.
+    /// @return amount1 The amount of token1
+    function getAmount1ForLiquidity(
+        uint256 liquidityChunk
+    ) internal pure returns (uint256 amount1) {
+        uint160 lowPriceX96 = getSqrtRatioAtTick(liquidityChunk.tickLower());
+        uint160 highPriceX96 = getSqrtRatioAtTick(liquidityChunk.tickUpper());
+
+        return mulDiv96(liquidityChunk.liquidity(), highPriceX96 - lowPriceX96);
+    }
+
+    /// @notice Calculates the amount of token0 and token1 received for a given liquidityChunk at the provided currentTick
+    /// @param currentTick the current tick to be evaluated
+    /// @param liquidityChunk variable that efficiently packs the liquidity, tickLower, and tickUpper.
+    /// @return amount0 The amount of token0
+    /// @return amount1 The amount of token1
+    function getAmountsForLiquidity(
+        int24 currentTick,
+        uint256 liquidityChunk
+    ) internal pure returns (uint256 amount0, uint256 amount1) {
+        uint160 lowPriceX96 = getSqrtRatioAtTick(liquidityChunk.tickLower());
+        uint160 highPriceX96 = getSqrtRatioAtTick(liquidityChunk.tickUpper());
+
+        uint160 currentPriceX96 = getSqrtRatioAtTick(currentTick);
```

Comment:
Because we pass in the current tick (as opposed to the more accurate currentSqrtPriceX96), there is no benefit to doing the extra work and converting the ticks to sqrt prices + using them to determine ITM/ATM/OTM.  Since we only have tick-level precision on the arguments to this function, it would always yield the same result if we just used the ticks directly.

Reply:
Yes! Great point